lib/model.js

```diff
@@ -3052,35 +3052,19 @@ class Model {
     }
   }
 
-  // Inject _scope into options. Includes should have been conformed (conformOptions) before calling this
+  // Inject _scope into options.
   static _injectScope(options) {
     const scope = Utils.cloneDeep(this._scope);
 
-    const filteredScope = _.omit(scope, 'include'); // Includes need special treatment
-
-    Utils.defaults(options, filteredScope);
-    Utils.defaults(options.where, filteredScope.where);
+    Utils.defaults(options, scope);
+    Utils.defaults(options.where, scope.where);
 
+    // Includes need special treatment
     if (scope.include) {
-      options.include = options.include || [];
-
-      // Reverse so we consider the latest include first.
-      // This is used if several scopes specify the same include - the last scope should take precedence
-      for (const scopeInclude of scope.include.reverse()) {
```

Comment:
What is the new merge strategy now (wrt https://github.com/sequelize/sequelize/blob/master/docs/scopes.md#merging), as far as I can tell it is still the same?

Reply:
Yeah, it's almost the same. I just updated for issues #9087,#6245

For example, current merge strategy is as follows.

```
Company.addScope('scopeA', {
  where: { name: 'A', something: true },
  attributes: {
    include: ['name']
  },
  include: [
    {
      model: Project,
      where: { name: 'A', something: true },
    },
    Project
  ],
  limit: 5,
  order: [
    ['name', 'DESC'],
  ]
})

Company.addScope('scopeB', {
  where: { name: 'B' },
  attributes: {
    include: ['something']
  },
  include: [
    {
      model: Project,
      as: 'Pj'
    },
    {
      model: Project,
      where: { name: 'B' }
    },
  ],
  limit: 10,
  order: [
    ['something', 'ASC'],
  ]
})

expect(Company.scope(['scopeA', 'scopeB'])._scope).to.deep.equal({
  where: { name: 'B', something: true }, // 'where' is shallowly merged
  attributes: { // just override
    include: ['something']
  },
  include: [ // 'include' is shallowly merged with the same model and alias(as)
    {
      model: Project,
      where: { name: 'B', something: true },
    },
    {
      model: Project,
      as: 'Pj'
    },
  ],
  limit: 10, // just override
  order: [ // 'Array' is united
    ['name', 'DESC'],
    ['something', 'ASC']
  ]
})
```

Reply:
Hmm I thought we discussed to have `attributes.include / exclude` merge https://github.com/sequelize/sequelize/pull/9735#issuecomment-413088477

Reply:
Oh, i'm sorry. I fixed it. e31be36

```
Company.addScope('scopeA', {
  attributes: {
    include: ['name']
  }
})

Company.addScope('scopeB', {
  attributes: {
    include: ['something'],
    exclude: ['password']
  }
})

expect(Company.scope(['scopeA', 'scopeB'])._scope).to.deep.equal({
  attributes: { // merged
    include: ['name', 'something'],
    exclude: ['password']
  }
})
```